### ISSUE TITLE
Reconnecting to a VMI over VNC fails eventually

### ISSUE THREAD

**Issue by qmfrederik** [kubevirt/kubevirt#1838]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
When I repeatedly open and close a VNC connection to a VMI, the `virtctl vnc` command will eventually fail with this error message:

```
C:\Users\frede\kubevirt>virtctl.exe --kubeconfig=kube.config vnc android-x86
{"component":"virtctl","level":"info","msg":"connection timeout: 1m0s","pos":"vnc.go:119","timestamp":"2018-12-11T14:13:52.701231Z"}
{"component":"virtctl","level":"info","msg":"remote-viewer connected in 1.6116144s","pos":"vnc.go:130","timestamp":"2018-12-11T14:13:54.312846Z"}
Error encountered: websocket: close 1006 (abnormal closure): unexpected EOF
```

Additionally, the following is visible in the virt-api logs:

```
{"component":"virt-api","level":"info","msg":"http: response.WriteHeader on hijacked connection\n","pos":"server.go:2923","timestamp":"2018-12-11T13:10:59.910305Z"}
{"component":"virt-api","level":"error","msg":"error ecountered reading from websocket stream","pos":"subresource.go:127","reason":"read tcp 172.17.0.12:8443-\u003e192.168.10.212:40592: use of closed network connection","timestamp":"2018-12-11T13:10:59.910419Z"}
{"component":"virt-api","contentLength":0,"level":"info","method":"GET","pos":"filter.go:46","proto":"HTTP/1.1","remoteAddress":"192.168.10.212","statusCode":500,"timestamp":"2018-12-11T13:10:59.910476Z","url":"/apis/subresources.kubevirt.io/v1alpha2/namespaces/default/virtualmachineinstances/android-x86/vnc","username":"-"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57114: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:11:06.936874Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57158: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:11:16.936792Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57202: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:11:26.937071Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57246: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:11:36.938698Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57290: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:11:46.937115Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57342: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:11:56.937088Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57386: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:12:06.936802Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57430: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:12:16.937453Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57474: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:12:26.937391Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57518: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:12:36.937363Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57562: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:12:46.937485Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57614: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:12:56.936936Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57658: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:13:06.936936Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57702: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:13:16.936771Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57746: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:13:26.937243Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57790: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:13:36.937270Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57834: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:13:46.943957Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57894: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:13:56.937766Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57938: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:14:06.937294Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:57982: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:14:16.937314Z"}
{"component":"virt-api","level":"error","msg":"error ecountered reading from websocket stream","pos":"subresource.go:127","reason":"websocket: close 1006 (abnormal closure): unexpected EOF","timestamp":"2018-12-11T13:14:17.666120Z"}
{"component":"virt-api","level":"error","msg":"Error in websocket proxy","pos":"subresource.go:136","reason":"websocket: close 1006 (abnormal closure): unexpected EOF","timestamp":"2018-12-11T13:14:17.666369Z"}
{"component":"virt-api","level":"info","msg":"http: response.WriteHeader on hijacked connection\n","pos":"server.go:2923","timestamp":"2018-12-11T13:14:17.666811Z"}
{"component":"virt-api","contentLength":0,"level":"info","method":"GET","pos":"filter.go:46","proto":"HTTP/1.1","remoteAddress":"192.168.10.212","statusCode":500,"timestamp":"2018-12-11T13:14:17.667360Z","url":"/apis/subresources.kubevirt.io/v1alpha2/namespaces/default/virtualmachineinstances/android-x86/console","username":"-"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 172.17.0.1:58028: EOF\n","pos":"server.go:2923","timestamp":"2018-12-11T13:14:26.937337Z"}
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a VMI
2. Open and close VNC connections to that VMI
3. If needed, delete the VMI and create a new VMI with the same name (not sure if this is a required step)
4. Deleting the VMI and the `virt-api-*` pods helps to recover from this problem.

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`): v0.12.0-alpha.0
- Kubernetes version (use `kubectl version`): v1.13.0
- VM or VMI specifications: android-x86
- Cloud provider or hardware configuration: bare metal
- OS (e.g. from /etc/os-release): Ubuntu 18.04
- Kernel (e.g. `uname -a`): Linux qm-playground 4.17.0-18-kubedroid-host #1 SMP Mon Dec 10 14:36:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Others:

**Comment by fabiand** [comment]
@qmfrederik do you have an idea what could be going wrong?

**Comment by slintes** [comment]
Will close this, it is probably fixed by #2171 
@qmfrederik please reopen if you still run into this with the latest KubeVirt release, thanks!

/close
